include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -280,12 +270,74 @@ struct GemmKernel
                     sequence<GemmPipeline::kPadM, false>{});
             }
         }();
-        auto CBlockWindow_pad = make_tile_window(
+
+        return make_tuple(a_pad_view, b_pad_view, c_pad_view);
+    }
+
+    template <typename PadView>
+    CK_TILE_DEVICE auto
+    make_gemm_tile_windows(PadView&& views, const index_t i_m, const index_t i_n) const
+    {
+        auto a_pad_view     = views.at(I0);
+        auto a_block_window = make_tile_window(
+            a_pad_view,
+            make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kK>{}),
+            {i_m, 0});
+
+        auto b_pad_view     = views.at(I1);
+        auto b_block_window = make_tile_window(
+            b_pad_view,
+            make_tuple(number<TilePartitioner::kN>{}, number<TilePartitioner::kK>{}),
+            {i_n, 0});
+
+        auto c_pad_view           = views.at(I2);
+        const auto c_block_window = make_tile_window(
             c_pad_view,
             make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
             {i_m, i_n});
 
-        EpiloguePipeline{}(CBlockWindow_pad, c_block_tile);
+        return make_tuple(a_block_window, b_block_window, c_block_window);
+    }
+
+    CK_TILE_DEVICE void run_common_gemm_pipeline(const ADataType* a_start,
+                                                 const BDataType* b_start,
+                                                 CDataType* c_start,
+                                                 const GemmCommonKargs& kargs,
+                                                 const index_t i_m,
+                                                 const index_t i_n) const
+    {
+        // Convert pointers to tensor views
+        const auto gemm_tensor_views_tuple =
+            make_gemm_tensor_views(a_start, b_start, c_start, kargs);
+        const auto gemm_pad_views    = make_gemm_pad_views(gemm_tensor_views_tuple);
+        const auto gemm_tile_windows = make_gemm_tile_windows(gemm_pad_views, i_m, i_n);
```

Comment:
Does `const` applies here to all windows? What about `C` ?

Reply:
You are right, after looking through it I think const is not needed here